unittests/ranges/test_ranges.cpp

```diff
@@ -0,0 +1,497 @@
+        QCOMPARE(reply->readAll().size(), 0);
+    }
+
+    reply->deleteLater();
+}
+
+void TestRangeRequests::testMultipleRanges_data()
+{
+    QTest::addColumn<QString>("rangeHeader");
+    QTest::addColumn<QList<QByteArray>>("expectedRanges");
+    QTest::addColumn<int>("expectedStatusCode");
+
+    QTest::newRow("separate_non-overlapping") << "bytes=100-199,300-399"
+                                              << QList<QByteArray> {expectedRange(100, 100), expectedRange(300, 100)} << 206;
+
+    QTest::newRow("adjacent_should_coalesce") << "bytes=400-499,500-599"
+                                              << QList<QByteArray> {expectedRange(400, 200)} << 206;
+
+    QTest::newRow("overlapping_should_coalesce") << "bytes=600-700,650-750"
+                                                 << QList<QByteArray> {expectedRange(600, 151)} << 206;
+
+    QTest::newRow("gapped_should_stay_separate") << "bytes=800-850,860-900"
+                                                 << QList<QByteArray> {expectedRange(800, 51), expectedRange(860, 41)} << 206;
+
+    // Range covers full file, status can be 200 or 206, let's pick 206 and accept both in test
+    QTest::newRow("suffix_equal_to_file") << "bytes=-2048"
+                                          << QList<QByteArray> {expectedRange(0, 2048)} << 200;
+
+    QTest::newRow("unsatisfiable_range") << "bytes=3000-4000"
+                                         << QList<QByteArray> {} << 416;
+
+    QTest::newRow("separate_non-overlapping_includes_start") << "bytes=0-100,200-299"
+                                                             << QList<QByteArray> {expectedRange(0, 100), expectedRange(200, 100)} << 206;
+}
+
+void TestRangeRequests::testMultipleRanges()
+{
+    QFETCH(QString, rangeHeader);
+    QFETCH(QList<QByteArray>, expectedRanges);
+    QFETCH(int, expectedStatusCode);
+
+    QNetworkRequest request(testUrl);
+    request.setRawHeader("Range", rangeHeader.toUtf8());
+
+    QNetworkReply *reply = manager->get(request);
+    QSignalSpy spy(reply, &QNetworkReply::finished);
+    QVERIFY(spy.wait());
+
+    const int status = reply->attribute(QNetworkRequest::HttpStatusCodeAttribute).toInt();
+
+    if (expectedRanges.isEmpty()) {
+        QCOMPARE(status, 416);
+        QCOMPARE(status, expectedStatusCode);
+    } else if (expectedRanges.size() == 1) {
+        if (expectedStatusCode == 200 && status != 200) {
+            // Accept 200 or 206 when full content is returned for request
+            QCOMPARE(status, 206);
+        } else
+            QCOMPARE(status, expectedStatusCode);
+        const QByteArray contentTypeHeader = reply->header(QNetworkRequest::ContentTypeHeader).toByteArray();
+        QVERIFY(!contentTypeHeader.startsWith("multipart/byteranges"));
+        const QByteArray data = reply->readAll();
+        QCOMPARE(data, expectedRanges.first());
+    } else {
+        QCOMPARE(status, 206);
+        QCOMPARE(status, expectedStatusCode);
```

Comment:
How does this relate to the previous line?

Are we expecting 206 or expectedStatusCode, or are both always equal?

This feels a bit like a unittest for the unittest ;-)

(same on lines 251/252).

If they're always equal, I'd just do `QCOMPARE(status, expectedStatusCode);` globally (i.e. I'd trust the `_test()` method to have the proper expected values), with the 200-or-206 caveat (maybe the test() method could provide a list instead of a single value...).